src/main/java/org/isomorphism/util/TokenBucket.java

```diff
@@ -24,6 +26,31 @@
 public interface TokenBucket
 {
   /**
+   * Returns the capacity of this token bucket.  This is the maximum number of tokens that the bucket can hold at
+   * any one time.
+   *
+   * @return The capacity of the bucket.
+   */
+  long getCapacity();
+
+  /**
+   * Returns the current number of tokens in the bucket.  If the bucket is empty then this method will return 0.
+   *
+   * @return The current number of tokens in the bucket.
+   */
+  long getNumTokens();
+
+  /**
+   * Returns the amount of time in the specified time unit until the next group of tokens can be added to the token
+   * bucket.
+   *
+   * @see org.isomorphism.util.TokenBucket.RefillStrategy#getDurationUntilNextRefill(java.util.concurrent.TimeUnit)
+   * @param unit The time unit to express the return value in.
+   * @return The amount of time until the next group of tokens can be added to the token bucket.
+   */
+  long getDurationUntilNextRefill(TimeUnit unit) throws UnsupportedOperationException;
```

Comment:
Yes, my intention was that if there's a version of a refill strategy being used that for some reason cannot implement the `getDurationUntilNextRefill` method that it would throw `UnsupportedOperationException`.  I feel like I'd rather have cases like this be very explicit that calling the method is a problem vs silently returning some value.